k256/src/arithmetic/field.rs

```diff
@@ -420,9 +458,32 @@ impl Neg for &FieldElement {
     }
 }
 
+impl Sum for FieldElement {
+    fn sum<I: Iterator<Item = Self>>(iter: I) -> Self {
+        iter.fold(Self::ZERO, |acc, w| acc + w)
```

Comment:
`iter.reduce(core::ops::Add::add).unwrap_or(Self::ZERO)` makes it use one less addition. Same for the others.

Reply:
Pushed improved `Sum`/`Product` impls in https://github.com/RustCrypto/elliptic-curves/pull/730/commits/d26419152b348fb13aa2338bb0b1107165218aff.

I used `.copied()` for when `Item` is borrowed... not sure if you had a better solution than that.

Reply:
I used `cloned()` which I guess amounts to the same thing.

Reply:
Yeah saw your version now... nice calling the other impl. Did that in [a10b1b8](https://github.com/RustCrypto/elliptic-curves/pull/730/commits/a10b1b81b3cacb179cbdc1d8b39bf0f8b887e61d)